### PR TITLE
Formula inputs by flags

### PR DESCRIPTION
**Pull request by kaduartur**
**- What I did**
I allow the formula inputs by flags:
image

- When an input by the flag is empty the Ritchie calls the prompt to the user to insert this value.
- Edited the code owners 
- Created an input resolver for formula runner
- Split the inputs.go into `input/prompt, input/stdin, and input/flag` to make easier the maintainability.
- improved how to add reserved flags
- now you can use the input by stdin without the `--stdin` flag for formulas command.

**- How to verify it**
1 - Checkout in this branch
2 - Build Ritchie locally `make build-{your-operational system}` and replace you locally installation
3 - Show your formula flags with `rit formula name --help`, you can see the inputs name such as flags
4 - Run your formulas with flags

**- Description for the changelog**
Allow formula inputs with flags

### CODE REVIEW

File: .github/ritchie-bot-config.yml [hunk 1]
```diff
@@ -2,22 +2,15 @@ merge_branch:
   enable: true # Enable merge branch with comment
   branches: ['qa'] # Branches enabled to merge with comment
   users: [
-    'sandokandias',
-    'marcosgmgm',
     'kaduartur',
     'ernelio',
     'viniciusramosdefaria',
-    'guillaumefalourd',
```
Comment (GuillaumeFalourd) [inline .github/ritchie-bot-config.yml:10]: This account *GuillaumeFalourd* is corporative (ZupIT)

**Review by GuillaumeFalourd — CHANGES_REQUESTED** [review]

**Review by henriquemoraeszup — COMMENTED** [review]
Bravo 👏 still haven't read it all, but I left a suggestion for debate

File: pkg/cmd/formula.go [hunk 2]
```diff
@@ -144,13 +150,31 @@ func (f FormulaCommand) execFormulaFunc(repo, path string) func(cmd *cobra.Comma
...
+		if err != nil {
+			return err
+		}
+
+		inputType := api.Prompt
+		if stdin {
+			inputType = api.Stdin
+		} else if flags.NFlag() > flagsCount {
+			inputType = api.Flag
+		}
```
[the remark this answers is not in the document]
Comment (GuillaumeFalourd) [reply pkg/cmd/formula.go:176]: I think it makes sense 👍🏼 
I just don't remember when the `STDIN` was implemented why we used the flag and not the `|` detection.

[the remark this answers is not in the document]
Comment (kaduartur) [reply pkg/cmd/formula.go]: it has been defined that each type of input will have its own responsibility, we will remove the necessity to use the --stdin option for formulas and main commands

File: pkg/formula/input/prompt/prompt.go [hunk 3]
```diff
@@ -75,113 +75,61 @@ func NewInput(
...
+				inputVal, err = in.textValidator(i)
 			}
-		case "bool":
-			valBool, err = in.Bool(input.Label, items, input.Tutorial)
+		case input.BoolType:
+			valBool, err = in.Bool(i.Label, items, i.Tutorial)
 			inputVal = strconv.FormatBool(valBool)
-		case "password":
-			inputVal, err = in.Password(input.Label, input.Tutorial)
-		case "dynamic":
```
Comment (henriquemoraeszup) [inline pkg/formula/input/prompt/prompt.go:152]: removing magic strings, nice 👍

File: pkg/formula/input/flag/flag.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	if len(emptyInputs) > 0 {
+		var emptyFlags []string
+		for i := 0; i < len(emptyInputs); i++ {
+			if input.IsRequired(emptyInputs[i]) {
+				emptyFlags = append(emptyFlags, fmt.Sprintf("--%s", emptyInputs[i].Name))
+			}
+		}
+
+		return fmt.Errorf("this flags cannot be empty [%s]", strings.Join(emptyFlags, ", "))
+	}
```
[the remark this answers is not in the document]
Comment (kaduartur) [reply pkg/formula/input/flag/flag.go:77]: Good suggestion, thanks 🚀

**Review by henriquemoraeszup — APPROVED** [review]

File: .github/OWNERS [hunk 5]
```diff
@@ -1,25 +0,0 @@
...
-* @sandokandias
-* @marcosgmgm
-* @kaduartur
-* @ernelio
-* @viniciusramosdefaria
-* @guillaumefalourd
-* @rodrigomedeirosf
-* @victor-schumacher
-* @brunasilvazup
-* @lucasdittrichzup
```
Comment (brunasilvazup) [inline .github/OWNERS:13]: Shouldn't we keep the reviewers?

File: pkg/cmd/formula.go [hunk 6]
```diff
@@ -169,9 +215,71 @@ func (f FormulaCommand) execFormulaFunc(repo, path string) func(cmd *cobra.Comma
...
+	}
+
+	return stat.Size() > 0
+}
+
+func isInputFlag(cmd *cobra.Command) bool {
+	flags := cmd.Flags()
+	c := 0
+	for _, flag := range reservedFlags {
+		if changed := flags.Changed(flag.name); changed {
```
Comment (brunasilvazup) [inline pkg/cmd/formula.go:274]: Here you could just leave the `flags.Changed(flag.name)`, what do you think?

File: pkg/cmd/formula.go [hunk 7]
```diff
@@ -169,9 +215,71 @@ func (f FormulaCommand) execFormulaFunc(repo, path string) func(cmd *cobra.Comma
...
+	for _, flag := range reservedFlags {
+		switch flag.kind {
+		case reflect.String:
+			flags.StringP(flag.name, flag.shortName, flag.defValue.(string), flag.description)
+		case reflect.Bool:
+			flags.BoolP(flag.name, flag.shortName, flag.defValue.(bool), flag.description)
+		case reflect.Int:
+			flags.IntP(flag.name, flag.shortName, flag.defValue.(int), flag.description)
+		default:
+			prompt.Warning("this type of flag is not supported")
```
Comment (brunasilvazup) [inline pkg/cmd/formula.go:228]: Could you add the name of the flag and its type, to give more information, what do you think?

File: pkg/cmd/formula.go (hunk 6 above)
Comment (kaduartur) [reply pkg/cmd/formula.go:274]: OMG, thanks for this review LOL

File: .github/OWNERS (hunk 5 above)
Comment (kaduartur) [reply .github/OWNERS:13]: this file doesn't work for Github, I created a CODEOWNERS file.

File: pkg/cmd/formula.go (hunk 7 above)
Comment (kaduartur) [reply pkg/cmd/formula.go:228]: Nice! 👀

**Review by victorschumacherzup — APPROVED** [review]

**Review by victorschumacherzup — APPROVED** [review]

**Review by lucasdittrichzup — DISMISSED** [review]

File: pkg/cmd/formula.go [hunk 8]
```diff
@@ -169,9 +215,77 @@ func (f FormulaCommand) execFormulaFunc(repo, path string) func(cmd *cobra.Comma
...
 }
 
-func addFlags(cmd *cobra.Command) {
-	formulaFlags := cmd.Flags()
-	formulaFlags.BoolP(formula.DockerRun.String(), "d", false, "Use to run formulas inside docker")
-	formulaFlags.BoolP(formula.LocalRun.String(), "l", false, "Use to run formulas locally")
-	formulaFlags.BoolP(verboseFlag, "a", false, "Verbose mode (All). Indicate to a formula that it should show log messages in more detail")
+func addReservedFlags(flags *pflag.FlagSet) {
+	for _, flag := range reservedFlags {
+		switch flag.kind {
```
Comment (victorschumacherzup) [inline pkg/cmd/formula.go:220]: nice

**Review by victorschumacherzup — APPROVED** [review]

**Review by lucasdittrichzup — APPROVED** [review]

**Pull request merged**